src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -2374,6 +2398,310 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
                 switch (intrinsic)
                 {
 #if defined(TARGET_XARCH)
+                    case NI_AVXVNNIINT_MultiplyWideningAndAdd:
```

Comment:
Should these be special imports instead?

This is quite a lot of code to put into the main table driven path.

Reply:
It seems like some of this could be simplified as well by using the `BaseTypeFromSecondArg` switch, right?

That way we only have to special case ID selection when `op2` is signed